tokio/src/runtime/context.rs

```diff
@@ -85,6 +88,18 @@ cfg_rt! {
 
     pub(crate) struct DisallowBlockInPlaceGuard(bool);
 
+    pub(crate) fn set_current_task_id(id: Option<Id>) {
+        CONTEXT.with(|ctx| ctx.current_task_id.replace(id));
+    }
```

Comment:
This will panic if it runs during the destructor of a thread-local. I would silently do nothing here so that destructors can still run even if thread-locals are not available.

Some sort of test for this would be nice as well, but there are already existing bugs related to this (see #5162), and you don't have to fix those in this PR, so you should only add such a test if it doesn't hit an existing bug about this.